CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -181,6 +197,33 @@ from UserCandies
             }
         }
 
+        // ConsumeRandomCandy: finds UserCandyId for oldest piece of a random candy by flavor and calls EatCandy
+        public UserCandyDetailed ConsumeRandomCandy(string flavor, int userId)
+        {
+            var sql = @"select top 1 with ties UserCandyId
+                        from UserCandies uc
+                            join Candies c
+                            on uc.candyId = c.candyId
+                        where isConsumed = 0
+                        and flavorCategory = @Flavor
```

Comment:
This should be @FlavorCategory instead of @Flavor

Reply:
updated variable names flavor/Flavor to flavorCategory/FlavorCategory